app/src/main/java/com/example/interntasks_2a/Case1/WeatherAdapter.kt

```diff
@@ -0,0 +1,73 @@
+package com.example.interntasks_2a.Case1
+
+import android.content.Context
+import android.content.Intent
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.core.content.ContextCompat.startActivity
+import androidx.recyclerview.widget.RecyclerView
+import com.example.interntasks_2a.R
+
+class WeatherAdapter(private val weatherList: List<Weather>) :
+    RecyclerView.Adapter<WeatherAdapter.WeatherViewHolder>() {
+
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int
+    ): WeatherAdapter.WeatherViewHolder {
+        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_weather, parent, false)
+        return WeatherViewHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: WeatherAdapter.WeatherViewHolder, position: Int) {
+        val weather = weatherList[position]
+        holder.bind(weather)
+    }
+
+    override fun getItemCount(): Int {
+        return weatherList.size
+    }
+
+    class WeatherViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
```

Comment:
Bu task özelinde gerekli değil ama bundan sonrası için `viewBinding` konusuna bakalım. Özellike `viewBinding`'in adapter içinde kullanımı. Sonraki task'larda alışkanlık haline getirebiliriz.

Reply:
Unuttuğumu şuan fark ettim activitylerimde viewBinding mevcut. Buraya da hemen ekliyorum, çok teşekkürler.